stubs/CMakeLists.txt

```diff
@@ -35,57 +25,18 @@ ENDIF()
 # no overflow warnings because of silly coin-ness
 SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-overflow")
 
-# Direct any out-of-source builds to this directory
-SET(STUB_SOURCE_DIR ${CMAKE_SOURCE_DIR})
-
 # Direct any binary installation paths to this directory
 SET(STUB_BINARY_DIR ${CMAKE_BINARY_DIR})
 
 # This makes all the libraries build as SHARED
 SET(BUILD_SHARED_LIBS true)
 
 # Setup build locations.
-IF(NOT CMAKE_RUNTIME_OUTPUT_DIRECTORY)
-    SET(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${STUB_BINARY_DIR}/bin)
-endif()
-IF(NOT CMAKE_LIBRARY_OUTPUT_DIRECTORY)
-    SET(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${STUB_BINARY_DIR}/lib)
-endif()
-IF(NOT CMAKE_ARCHIVE_OUTPUT_DIRECTORY)
-    SET(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${STUB_BINARY_DIR}/lib)
-ENDIF()
+cyclus_setup_build_locations(${STUB_BINARY_DIR})
 
 SET(STUB_EXECUTABLE_DIR ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
 
-# use, i.e. don't skip the full RPATH for the build tree
-SET(CMAKE_SKIP_BUILD_RPATH  FALSE)
-
-# when building, don't use the install RPATH already
-# (but later on when installing)
-SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
-
-SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
-
-# add the automatically determined parts of the RPATH
-# which point to directories outside the build tree to the install RPATH
-SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-
-# the RPATH to be used when installing, but only if it's not a system directory
-LIST(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_PREFIX}/lib" isSystemDir)
-IF("${isSystemDir}" STREQUAL "-1")
-    SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib:${CMAKE_INSTALL_PREFIX}/lib/cyclus")
-    GET_FILENAME_COMPONENT(cxxCompilerRoot ${CMAKE_CXX_COMPILER} DIRECTORY)
-    GET_FILENAME_COMPONENT(cxxCompilerRoot ${cxxCompilerRoot} DIRECTORY)
-    IF (NOT "${CMAKE_INSTALL_RPATH}" STREQUAL "${cxxCompilerRoot}")
-        SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_RPATH}:${cxxCompilerRoot}/lib:${cxxCompilerRoot}/lib/cyclus")
-    ENDIF (NOT "${CMAKE_INSTALL_RPATH}" STREQUAL "${cxxCompilerRoot}")
-    SET(CMAKE_BUILD_RPATH "${CMAKE_INSTALL_RPATH}")
-ENDIF("${isSystemDir}" STREQUAL "-1")
-
-# Tell CMake where the modules are
-SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_INSTALL_PREFIX}/share/cyclus/cmake")
-SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${STUB_SOURCE_DIR}/cmake)
-MESSAGE("--CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH}")
+cyclus_set_rpath()
 
 # Find cyclus
 FIND_PACKAGE(Cyclus REQUIRED)
```

Comment:
If we move this to the top... could we use it to set the `CMAKE_MODULE_PATH` more robustly?

Reply:
> could we use it to set the `CMAKE_MODULE_PATH` more robustly?

I tried moving line 42 to the top but I'm pretty sure we need to set the CMAKE_MODULE_PATH in order for cmake to find our `FindCyclus.cmake` file in `$HOME/.local/share/cyclus/cmake`

This is the log from `test_stubs.py`:
```bash
/usr/bin/cmake
CMake Error at CMakeLists.txt:4 (FIND_PACKAGE):
  No "FindCyclus.cmake" found in CMAKE_MODULE_PATH.


CMake Warning (dev) at CMakeLists.txt:4 (FIND_PACKAGE):
  FindCyclus.cmake must either be part of this project itself, in this case
  adjust CMAKE_MODULE_PATH so that it points to the correct location inside
  its source tree.

  Or it must be installed by a package which has already been found via
  find_package().  In this case make sure that package has indeed been found
  and adjust CMAKE_MODULE_PATH to contain the location where that package has
  installed FindCyclus.cmake.  This must be a location provided by that
  package.  This error in general means that the buildsystem of this project
  is relying on a Find-module without ensuring that it is actually available.

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring incomplete, errors occurred!
Traceback (most recent call last):
  File "/tmp/tmpyubn5vng/foo/install.py", line 137, in <module>
    main()
  File "/tmp/tmpyubn5vng/foo/install.py", line 134, in main
    install(args)
  File "/tmp/tmpyubn5vng/foo/install.py", line 61, in install
    rtn = subprocess.check_call(cmake_cmd, cwd=args.build_dir,
  File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '/tmp/tmpyubn5vng/foo', '-DCMAKE_INSTALL_PREFIX=/tmp/tmpyubn5vng/baz', '-DCYCLUS_ROOT_DIR=/home/ben/.local']' returned non-zero exit status 1.

```

Reply:
Hmm... you're right though.  I wonder if we should copy a tiny bit of logic from FindCyclus.cmake so that we have more confidence in where the cyclus install is (call `which cyclus`)